### PR TITLE
Gdr 2002

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Switch from `aggregate` to `data.table`
Related JIRA issue: GDR-2002

## Why was it changed?
To switch to data.table solution

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

File: R/fit_curves.R [hunk 1]
```diff
@@ -511,17 +511,9 @@ has_dups <- function(vec) {
 
 #' @keywords internal
 #' @importFrom stats aggregate
```
Comment (j-smola) [inline R/fit_curves.R:513]:
```suggestion
```

File: R/fit_curves.R [hunk 2]
```diff
@@ -511,17 +511,9 @@ has_dups <- function(vec) {
...
-    aggregate(df[, agg_col, with = FALSE],
-              by = list(concs = df[[col]]),
-              FUN = function(x) {
-                mean(x, na.rm = TRUE)
-              }
-    )
-  )
+average_dups <- function(dt, col) {
+  stopifnot(col %in% colnames(dt))
+  dt[, lapply(.SD, mean, na.rm = TRUE), by = col]
```
[the remark this answers is not in the document]
Comment (bczech) [reply R/fit_curves.R:517]: Thanks, done in 7f42a9f

File: R/fit_curves.R (hunk 1 above)
Comment (bczech) [reply R/fit_curves.R]: Done

**Review by j-smola — APPROVED** [review]

File: R/fit_curves.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (bczech) [reply R/fit_curves.R:516]: @gladkia I changed the logic in 6a941eb since we assume that all columns will be numeric for usage of this fun

**Review by gladkia — APPROVED** [review]

**Pull request merged**